src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            return (List<T>) new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            return (List<T>) new TruckProducer().get();
+        }
+        return new ArrayList<>();
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. Directly returning the result of `new BulldozerProducer().get()` is not recommended, as it does not allow for easy debugging or extension of the method's functionality. Instead, consider creating a local variable to hold the result before returning it.